pkg/controllers/core/catalog_controller_test.go

```diff
@@ -539,9 +540,264 @@ func TestCatalogdControllerReconcile(t *testing.T) {
 			} else {
 				assert.Error(t, err)
 			}
-
 			diff := cmp.Diff(tt.expectedCatalog, tt.catalog, cmpopts.IgnoreFields(metav1.Condition{}, "Message", "LastTransitionTime"))
 			assert.Empty(t, diff, "comparing the expected Catalog")
 		})
 	}
 }
+
+func TestPollingRequeue(t *testing.T) {
+	for name, tc := range map[string]struct {
+		catalog              *catalogdv1alpha1.Catalog
+		expectedRequeueAfter time.Duration
+	}{
+		"Catalog with tag based image ref without any poll interval specified, requeueAfter set to 0, ie polling disabled": {
+			catalog: &catalogdv1alpha1.Catalog{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "test-catalog",
+					Finalizers: []string{fbcDeletionFinalizer},
+				},
+				Spec: catalogdv1alpha1.CatalogSpec{
+					Source: catalogdv1alpha1.CatalogSource{
+						Type: "image",
+						Image: &catalogdv1alpha1.ImageSource{
+							Ref: "someimage:latest",
+						},
+					},
+				},
+			},
+			expectedRequeueAfter: time.Second * 0,
+		},
+		"Catalog with tag based image ref with poll interval specified, requeueAfter set to wait.jitter(pollInterval)": {
+			catalog: &catalogdv1alpha1.Catalog{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "test-catalog",
+					Finalizers: []string{fbcDeletionFinalizer},
+				},
+				Spec: catalogdv1alpha1.CatalogSpec{
+					Source: catalogdv1alpha1.CatalogSource{
+						Type: "image",
+						Image: &catalogdv1alpha1.ImageSource{
+							Ref:          "someimage:latest",
+							PollInterval: &metav1.Duration{Duration: time.Minute * 5},
+						},
+					},
+				},
+			},
+			expectedRequeueAfter: time.Minute * 5,
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			reconciler := &CatalogReconciler{
+				Client: nil,
+				Unpacker: source.NewUnpacker(
+					map[catalogdv1alpha1.SourceType]source.Unpacker{
+						catalogdv1alpha1.SourceTypeImage: &MockSource{result: &source.Result{
+							State: source.StateUnpacked,
+							FS:    &fstest.MapFS{},
+						}},
+					},
+				),
+				Storage: &MockStore{},
+			}
+			res, _ := reconciler.reconcile(context.Background(), tc.catalog)
+			assert.GreaterOrEqual(t, res.RequeueAfter, tc.expectedRequeueAfter)
+			assert.LessOrEqual(t, tc.expectedRequeueAfter, wait.Jitter(res.RequeueAfter, requeueJitterMaxFactor))
```

Comment:
My understanding is this is attempting to assert `tc.expectedRequeueAfter <= res.RequeueAfter <= jitter(tc.expectedRequeueAfter)`. Instead, the combination of these statements seem to be asserting
`tc.expectedRequeueAfter <= res.RequeueAfter , tc.expectedRequeueAfter <= jitter(res.RequeueAfter)`.

To properly test the first case, I think the position of the `tc.expectedRequeueAfter` and `res.RequeueAfter` would need to be flipped:
```suggestion
			assert.LessOrEqual(t, res.RequeueAfter, wait.Jitter(tc.expectedRequeueAfter, requeueJitterMaxFactor))
```

Reply:
Okay I actually finally sat down and thought about this. It has to be `assert.LessOrEqual(t, float64(res.RequeueAfter), float64(tc.expectedRequeueAfter)+(float64(tc.expectedRequeueAfter)*requeueJitterMaxFactor))` to be accurate. I think that's what Steve was trying to allude to too. Pushed the change.